DROD/SettingsScreen.cpp

```diff
@@ -1488,6 +1515,22 @@ void CSettingsScreen::SetWidgetStates()
 	pButton->Enable(!holdIds.empty());
 	if (bButtonsVisible)
 		pButton->RequestPaint();
+
+	SetCloudWidgetStates();
+}
+
+//************************************************************************************
+void CSettingsScreen::SetCloudWidgetStates()
+//Enable/disable widgets depending on state. Separate from SetWidgetStates due to how
+//cloud settings are handled.
+{
+	CDbPlayer* pPlayer = g_pTheDB->GetCurrentPlayer();
+
+	CButtonWidget* pButton = DYN_CAST(CButtonWidget*, CWidget*,
+		GetWidget(TAG_CLOUD_DEACTIVATE));
+	pButton->Enable(pPlayer->Settings.GetVar(Settings::CloudActivated, false));
```

Comment:
Should this setting be flipped? i.e., when CloudActivated is true, the deactivated button should be set to false, and v.v.

Reply:
The deactivate button is the button that deactivates the cloud sync, so we want it to be enabled when cloud sync is on, and off otherwise.
Maybe we want the activate button to enable/disable based on the profile setting as well.

Reply:
Ohhh...I get it now. I was thinking this was a check box (toggle) button, not a regular button.

How about instead of an "activate" and "deactivate" button, we just use one check box to drive both actions? I believe they are mutually exclusive, correct?

Reply:
I'll have a look about that, although I'm not sure if it will mesh with the use of dialog boxes.

Reply:
Good point. I'm wondering about this too. I'm pretty sure that when the box changes state, we can pop up the dialog box as before, and if the player doesn't confirm the action, we can revert the checkbox state.